temporal-sdk/src/main/java/io/temporal/internal/replay/ReplayWorkflowRunTaskHandler.java

```diff
@@ -188,11 +188,6 @@ private void handleWorkflowTaskImpl(PollWorkflowTaskQueueResponseOrBuilder workf
           implementationOptions.getFailWorkflowExceptionTypes();
       for (Class<? extends Throwable> failType : failTypes) {
         if (failType.isAssignableFrom(e.getClass())) {
-          // Wrap any failure into InternalWorkflowTaskException to support specifying them
```

Comment:
This piece doesn't make any sense to me and it makes unnecessary misleading wraps in the reported error.
@mfateev could you please check if you see big reasons for this code segment to stay?
InternalWorkflowTaskException is not any special and doesn't have special handling anywhere, so this wrapping is just wrapping.